### PR TITLE
Implement no_block for service.running/service.dead (systemd only)

### LINKED ISSUE #39570: Allow (systemd) services to be started non-blocking
I'd like to be able to tell salt that a specific service should be started, but that I do not want to wait for startup to be successful. This is equivalent to specifying `--no-block` on the systemctl command line. Essentially, something like the following.

```
my-service:
  service.running:
    - enable: true
    - no_block: true
```

Both code and docs have no mention of this being possible currently.

Let me add some context: I have a service for which start-up can take minutes / hours, since it needs to fetch a large amount of state via the network. I'd like to use systemd sd_notify and service dependencies to automatically start other services once the state fetch has completed.

As it currently stands, a highstate or salt-call will block until the service has started, which basically means our machines will go unresponsive. If I can opt out of the blocking behaviour I could get the best of both worlds.

I'm happy to contribute the code if you can point me to the correct approach.

### PR DESCRIPTION
**Pull request by terminalmage**
Resolves #39570.

### CODE REVIEW

File: salt/modules/systemd.py [hunk 1]
```diff
@@ -973,9 +1013,9 @@ def disable(name, **kwargs):  # pylint: disable=unused-argument
                                        python_shell=False,
                                        ignore_retcode=True) == 0
     return __salt__['cmd.retcode'](
-        _systemctl_cmd('disable', name, systemd_scope=True),
+        _systemctl_cmd('disable', name, systemd_scope=True, no_block=no_block),
         python_shell=False,
-        ignore_recode=True) == 0
```
Comment (lmb) [inline salt/modules/systemd.py]: Seems like this is an unrelated bugfix? Might be worth calling that out?

Comment (terminalmage) [inline salt/modules/systemd.py]: I don't know what you mean by "calling that out".

If you mean "mention it in the PR/commit msg", I don't think that's necessary since I opened https://github.com/saltstack/salt/pull/39633 to fix this in the 2016.3 release branch. I only made the change here because I also added a missing unit test, and I needed this fix here to make the test pass.

**Comment by lmb** [comment]
Great! Just checking :)

**Comment by terminalmage** [comment]
Cool. Thanks for reviewing the PR, by the way. :smile:

**Comment by cachedout** [comment]
@terminalmage Could you rebase this please? Merge conflict.

**Comment by terminalmage** [comment]
Rebased.
